### PR TITLE
KTOR-7679 Allow disabling body decoding on server

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-7679](https://youtrack.jetbrains.com/issue/KTOR-7679) Allow to Disable Body Decompression on the Server for a specific call

### CODE REVIEW

File: ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Compression.kt [hunk 1]
```diff
@@ -91,6 +75,10 @@ public val Compression: RouteScopedPlugin<CompressionConfig> = createRouteScoped
 @OptIn(InternalAPI::class)
 private fun decode(call: PipelineCall, options: CompressionOptions) {
     val encodingRaw = call.request.headers[HttpHeaders.ContentEncoding]
+    if (call.isCompressionSuppressed) {
+        LOGGER.trace("Skip decompression for ${call.request.uri} because it is suppressed.")
```
Comment (marychatte) [inline ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Compression.kt:79]: Shouldn't it be `call.isDecompressionSuppressed`

File: ktor-server/ktor-server-tests/jvm/test/io/ktor/server/plugins/CompressionTest.kt [hunk 2]
```diff
@@ -769,6 +769,54 @@ class CompressionTest {
...
+        val response = client.post("/gzip") {
+            setBody(compressed)
+            header(HttpHeaders.ContentEncoding, "gzip")
+            header(HttpHeaders.AcceptEncoding, "gzip")
+        }
+        assertContentEquals(textToCompressAsBytes, response.body<ByteArray>())
+    }
+
+    @Test
+    fun testDisableCallDecoding() = testApplication {
```
Comment (marychatte) [inline ktor-server/ktor-server-tests/jvm/test/io/ktor/server/plugins/CompressionTest.kt:797]: Why function `suppressDecompression()` is not used?

**Comment by e5l** [comment]
Hey, @marychatte! Thank you for the review. I've spotted other issue, let me check

File: ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Compression.kt (hunk 1 above)
Comment (e5l) [reply ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Compression.kt:81]: fixed

File: ktor-server/ktor-server-tests/jvm/test/io/ktor/server/plugins/CompressionTest.kt (hunk 2 above)
Comment (e5l) [reply ktor-server/ktor-server-tests/jvm/test/io/ktor/server/plugins/CompressionTest.kt:809]: fixed

**Comment by e5l** [comment]
Hey, @marychatte! Sorry for the delay, there was an issue with body decompression.
Ready to check!

**Review by marychatte — APPROVED** [review]
lgtm 👍

**Pull request merged**